src/backend/storeManagers/hyperplay/games.ts

```diff
@@ -1397,6 +1399,47 @@ export async function launch(appName: string): Promise<boolean> {
   return launchGame(appName, getGameInfo(appName), 'hyperplay')
 }
 
+async function createSiweMessage(signerAddress: string): Promise<SiweMessage> {
+  const mainWindowUrl = getMainWindow()?.webContents.getURL()
+  if (mainWindowUrl === undefined) {
+    throw 'could not get main window url'
+  }
+  const url = new URL(mainWindowUrl)
+  const domain = url.host ? url.host : 'hyperplay'
+  const origin = url.origin.startsWith('file://')
+    ? 'file://hyperplay'
+    : url.origin
+
+  const statementRes = await fetch(
+    DEV_PORTAL_URL + 'api/v1/license_contracts/validate/get-nonce'
+  )
+  const statement = String(statementRes?.body)
```

Comment:
Should this be `await res.text()` instead since it's fetch?

Reply:
good call. I added some better error handling here as well